src/Catalyst.Core.Modules.Dfs/BlockExchange/Protocols/Bitswap11.cs

```diff
@@ -0,0 +1,273 @@
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs.BlockExchange;
+using Catalyst.Abstractions.Dfs.BlockExchange.Protocols;
+using Common.Logging;
+using Lib.P2P;
+using MultiFormats;
+using ProtoBuf;
+using Semver;
+using ProtoBufHelper = Lib.P2P.ProtoBufHelper;
+
+#pragma warning disable 0649 // disable warning about unassinged fields
+#pragma warning disable 0169// disable warning about unassinged fields
+
+namespace Catalyst.Core.Modules.Dfs.BlockExchange.Protocols
+{
+    /// <summary>
+    ///   Bitswap Protocol version 1.1.0 
+    /// </summary>
+    public class Bitswap11 : IBitswapProtocol
+    {
+        private static ILog log = LogManager.GetLogger(typeof(Bitswap11));
+
+        /// <inheritdoc />
+        public string Name { get; } = "ipfs/bitswap";
+
+        /// <inheritdoc />
+        public SemVersion Version { get; } = new SemVersion(1, 1);
+
+        /// <inheritdoc />
+        public override string ToString() { return $"/{Name}/{Version}"; }
+
+        /// <summary>
+        ///   The <see cref="BitswapService"/> service.
+        /// </summary>
+        public IBitswapService BitswapService { get; set; }
+
+        /// <inheritdoc />
+        public async Task ProcessMessageAsync(PeerConnection connection,
+            Stream stream,
+            CancellationToken cancel = default)
+        {
+            // There is a race condition between getting the remote identity and
+            // the remote sending the first wantlist.
+            await connection.IdentityEstablished.Task.ConfigureAwait(false);
+
+            while (true)
+            {
+                var request = await ProtoBufHelper.ReadMessageAsync<Message>(stream, cancel).ConfigureAwait(false);
+
+                // Process want list
+                if (request.wantlist != null && request.wantlist.entries != null)
+                {
+                    foreach (var entry in request.wantlist.entries)
+                    {
+                        var cid = Cid.Read(entry.block);
+                        if (entry.cancel)
+                        {
+                            // TODO: Unwant specific to remote peer
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)